crates/metadata/src/tests.rs

```diff
@@ -28,11 +28,11 @@ fn spec_constructor_selector_must_serialize_to_hex() {
         .selector(123_456_789u32.to_be_bytes())
         .done();
     let mut registry = Registry::new();
-    let compact_spec = cs.into_compact(&mut registry);
+    let portable_spec = cs.into_portable(&mut registry);
 
     // when
-    let json = serde_json::to_value(&compact_spec).unwrap();
-    let deserialized: ConstructorSpec<CompactForm> =
+    let json = serde_json::to_value(&portable_spec).unwrap();
+    let deserialized: ConstructorSpec<PortableForm<String>> =
```

Comment:
Why `PortableForm<String>`? In which way is the `String` Rust type significantly involved with any of the serialization or deserialization? This seems to be a weak spot in the recent design changes where I have had some blind spot.

Reply:
Because String implements `DeserializeOwned`, whereas the default is `&'static str`. The driver behind the whole thing is that currently substrate runtimes are not allowed Strings, and to that end `parity-scale-codec` only supports them in `std`.

I am open to alternatives to solve the problem with a better API.

Reply:
conclusion from chat: we will open another issue on the `scale-info` crate to resolve this API design issue and will continue merging this PR.